harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -619,32 +1001,69 @@ def _compute_validation_metrics(self) -> workload.Metrics:
         ):
             logging.debug(
                 "Broadcasting metrics to all worker processes to execute a "
-                "validation step end callback"
+                "validation step end callback."
             )
             metrics = self.context.distributed.broadcast(metrics)
 
         for callback in self.callbacks.values():
             if util.is_overridden(callback.on_validation_step_end, pytorch.PyTorchCallback):
                 logging.warning(
-                    "on_validation_step_end is now deprecated, please use on_validation_end instead"
+                    "on_validation_step_end is now deprecated, please use on_validation_end instead."
                 )
                 callback.on_validation_step_end(metrics)
 
         for callback in self.callbacks.values():
             callback.on_validation_end(metrics)
 
-        if not self.is_chief:
-            return {}
+        self.state.last_val = self.state.batches_trained
+
+        # Report metrics
+        if self.is_chief:
+            # Skip reporting timings if evaluate_full_dataset() was defined.  This is far less
+            # common than evaluate_batch() and we can't know how the user processed their
+            # validation data.
+            if self._evaluate_batch_defined():
+                step_duration = time.time() - step_start_time
+                logging.info(
+                    det.util.make_timing_log("validated", step_duration, num_inputs, num_batches)
+                )
+            self.metric_writer.on_validation_step_end(self.state.batches_trained, metrics)
+            self.core_context.train.report_validation_metrics(self.state.batches_trained, metrics)
 
-        # Skip reporting timings if evaluate_full_dataset() was defined.  This is far less common
-        # than evaluate_batch() and we can't know how the user processed their validation data.
-        if self._evaluate_batch_defined():
-            step_duration = time.time() - step_start_time
-            logging.info(
-                det.util.make_timing_log("validated", step_duration, num_inputs, num_batches)
-            )
-        self.metric_writer.on_validation_step_end(self.steps_completed, metrics)
-        return {"num_inputs": num_inputs, "validation_metrics": metrics}
+        searcher_metric = None
+
+        # Report searcher status
+        if self.is_chief and searcher_op:
+            if self.local_training:
+                searcher_length = self.max_length
+            else:
+                searcher_length = TrainUnit._from_searcher_unit(
+                    searcher_op.length, self.searcher_unit, self.global_batch_size
+                )
+            if self.searcher_metric_name:
+                searcher_metric = self._check_searcher_metric(metrics)
```

Comment:
In local mode, don't you not have a searcher metric?  So won't this... not work?

Reply:
nvm, this is fine, Anda explained it to me.